src/code/events.asm

```diff
@@ -505,44 +505,57 @@ CheckKillInOrderTrigger::
 
 ; Check for both TRIGGER_KILL_ALL_ENEMIES and TRIGGER_KILL_SPECIALS
 CheckKillEnemiesTrigger::
-    ld   c, $0F                                   ; $5FFC: $0E $0F
+    ld   c, MAX_ENTITIES - 1                      ; $5FFC: $0E $0F
     ld   b, $00                                   ; $5FFE: $06 $00
 
-jr_002_6000:
+    ; Enumerate all entities, from $0F to 0
+.loopOverEntities
     ld   hl, wEntitiesStatusTable                 ; $6000: $21 $80 $C2
     add  hl, bc                                   ; $6003: $09
     ld   a, [hl]                                  ; $6004: $7E
+
+    ; If the entity is none or disabled, continue with the next entity
     and  a                                        ; $6005: $A7
-    jr   z, .jr_6011                              ; $6006: $28 $09
+    jr   z, .continue                             ; $6006: $28 $09
 
+    ; If the entity is active, and not excluded from the "Kill all" trigger,
+    ; then the trigger is not resolved: return immediately.
     ld   hl, wEntitiesOptions1Table               ; $6008: $21 $30 $C4
     add  hl, bc                                   ; $600B: $09
     ld   a, [hl]                                  ; $600C: $7E
     and  ENTITY_OPT1_EXCLUDED_FROM_KILL_ALL       ; $600D: $E6 $02
-    jr   z, ret_002_602C                          ; $600F: $28 $1B
+    jr   z, .return                               ; $600F: $28 $1B
 
-.jr_6011
+    ; Loop until all entities are enumerated
+.continue
     dec  c                                        ; $6011: $0D
     ld   a, c                                     ; $6012: $79
-    cp   $FF                                      ; $6013: $FE $FF
-    jr   nz, jr_002_6000                          ; $6015: $20 $E9
+    cp   -1                                       ; $6013: $FE $FF
+    jr   nz, .loopOverEntities                    ; $6015: $20 $E9
 
+    ; We enumerated all entities, and all of them are either disabled
+    ; or excluded from the "Kill all" trigger: the trigger might be
+    ; resolved.
+
+    ; If hMultiPurpose0 != 08, then the trigger is resolved.
     ldh  a, [hMultiPurpose0]                      ; $6017: $F0 $D7
     cp   $08                                      ; $6019: $FE $08
-    jr   nz, .jr_6029                             ; $601B: $20 $0C
+    jr   nz, .triggerSatisfied                    ; $601B: $20 $0C
 
+    ; If wD460 == 0, don't mark the trigger as resolved.
     ld   a, [wD460]                               ; $601D: $FA $60 $D4
```

Comment:
These conditions are probably for the "Kill enemies in the right order" trigger in D2.

Reply:
Maybe, but that's a separate trigger, though?

Reply:
It's not the order. It's only referenced from `EntityFallHandler`. Which I assume handles an entity falling in a pit. I assume it's a "trigger" for the event, so it only happens when something is killed, and not all the time.

Reply:
Yup, and seems to be set only when the falling enemy is a `ENTITY_OPT1_EXCLUDED_FROM_KILL_ALL`. Hmm.

Reply:
Seeing the other comment, I guess it's related to the special kill trigger?